### PR TITLE
RavenDB-10903 Rename Testing.ParentProcessId to Embedded.ParentProcessId

### PR DESCRIPTION
**Pull request by IdanHaim**

### CODE REVIEW

File: src/Raven.Server/Config/Categories/EmbeddedConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+namespace Raven.Server.Config.Categories
+{
+    public class EmbeddedConfiguration : ConfigurationCategory
+    {
+        private int? _legacyParentProcessId;
+        private int? _parentProcessId;
+
+        [Description("Watch the parent process id and exit when it exited as well (for tests)")]
+        [DefaultValue(null)]
+        [ConfigurationEntry("Testing.ParentProcessId", ConfigurationEntryScope.ServerWideOnly)]
```
Comment (ppekrol) [inline src/Raven.Server/Config/Categories/EmbeddedConfiguration.cs:13]: I think that we should add support for multiple keys in configuration entry. We do not really care from which key it was set probably or drop 'Testing.ParentProcessId' entirely.

Comment (ayende) [reply src/Raven.Server/Config/Categories/EmbeddedConfiguration.cs:13]: Good idea

Comment (IdanHaim) [reply src/Raven.Server/Config/Categories/EmbeddedConfiguration.cs:13]: Fixed

Comment (ayende) [reply src/Raven.Server/Config/Categories/EmbeddedConfiguration.cs:13]: No, we want a _single_ property, not LegacyParentProcessId and ParentProcessId

Comment (IdanHaim) [reply src/Raven.Server/Config/Categories/EmbeddedConfiguration.cs:13]: dropping the Testing.ParentProcessId?

**Pull request closed without merging**